src/main/resources/templates/fragments/header.html

```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE html>
+<html>
+<head>
+</head>
+  <body>
+    <div th:fragment="header">
+    	<nav class="navbar navbar-expand-sm bg-light">
+
+  <div class="container-fluid">
+    <h2>Gas Mileage Calculator</h2>
+  </div>
+
+</nav>
+    </div>
+  </body>
+</html>
```

Comment:
I made the navigation element / header into a thymeleaf fragment. This will really be beneficial once you start adding some more pages, as if you want to change the header component you will only have to change it once rather than having to change html in every template.

A good resource for thymeleaf page layouts, there is a section regarding fragments: https://www.thymeleaf.org/doc/articles/layouts.html